Sources/Sentry/SentryEnvelope.m

```diff
@@ -62,23 +63,30 @@ - (instancetype)initWithEvent:(SentryEvent *)event
     NSData *json = [SentrySerialization dataWithJSONObject:[event serialize] error:&error];
 
     if (nil != error) {
-        SentryEvent *errorEvent = [[SentryEvent alloc] initWithLevel:kSentryLevelWarning];
-
-        // Add some context to the event. We can only set simple properties otherwise we
-        // risk that the conversion fails again.
-        NSString *messge = [NSString
-            stringWithFormat:@"JSON conversion error for event with message: '%@'", event.message];
-        errorEvent.message = messge;
-        errorEvent.releaseName = event.releaseName;
-        errorEvent.environment = event.environment;
-        errorEvent.sdk = event.sdk;
-        errorEvent.platform = event.platform;
-        errorEvent.timestamp = event.timestamp;
-
-        // We accept the risk that this simple serialization fails which is covered by tests.
-        // Therefore we ignore the error on purpose and send an envelope item with an empty
-        // body.
-        json = [SentrySerialization dataWithJSONObject:[errorEvent serialize] error:nil];
+        // It could be the user added something to the context that can't be serialized. Try to
+        // serialize without the context.
+        event.context = nil;
+        error = nil;
+        json = [SentrySerialization dataWithJSONObject:[event serialize] error:&error];
+
+        // The context was the problem for serialization. Add a breadcrumb that we are dropping the
+        // context. We take the risk that another field causes the serialization error and ignore
+        // it.
+        if (nil == error) {
```

Comment:
M: should this `if` have an `else` which does the old behavior? creating an event like `JSON conversion error for event with message...` because right now it silently fails

Reply:
Currently, I can't even write a test for the `else` here, because I changed it to set both `context` and `sdk` to `nil`. For `extra` `sentry_sanitize` is used in `SentryEvent`. So I don't know how to set a value to a field that breaks the serialization. Therefore I just log a debug message so the users get the feedback when in debug. I rather not add the old behaviour for a very unlikely edge case.

Reply:
I'm fine how it is, but its a heads up, we closed a hole (now we know if its a broken context, not serializable), but we opened another one (if anything else is broken).

previously, we didn't know what it was broken, but we knew something was wrong.

I find it an improvement because the most common source of non-serializable errors is fixed, but on the other hand, when something happens in the wild, we'll not have a clue (we cant read logs on their phones).

Reply:
Good point 👏 , I changed it again. Please have another look @marandaneto .